### ISSUE TITLE
508-defect-2 [COGNITION]: Heading levels SHOULD increase by one

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#16692]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:**  Josh Kim

## User Story or Problem Statement

As a screen reader user, I want to navigate the hierarchy of the page content using heading levels to save time and frustration.

## Details

Headings on a web page should progress, increasing by one, in sections of content. There are a number of headings throughout the pages that don't adhere to the heading order of H1, H2, H3, H4, etc.

## Acceptance Criteria

Example:
- [ ] Headings progress one by one in a logical order

## Environment

* Operating System: all
* Browser: all
* Screenreading device: any
* Server destination: staging and production

## Steps to Recreate

1. [Open up the subway map page of VA Form 28-8832](https://staging.va.gov/careers-employment/education-and-career-counseling/apply-career-guidance-form-28-8832/introduction)
2. Have developer tools open, and the axe browser extension loaded
3. Run an axe audit
4. Verify that there is an error of "Heading levels should only increase by one"

## Solution (if known)

Update the headings into a logical order as demonstrated below. I have included visual heading classes to maintain the intended visual hierarchy.

### Before
```html
    <h3 class="usa-alert-heading">
      Save time—and save your work in progress—by signing in before starting your application
    </h3>

    <h4>
      Follow the steps below to apply for Planning and career guidance.
    </h4>

      <h5>
        Prepare
      </h5>

        <h6>
          To fill out this application, you’ll need your:
        </h6>

      <h5>
        Apply
      </h5>

      <h5>
        VA Review
      </h5>

      <h5>
        Decision
      </h5>

      <h5 class="vads-u-font-size--h3 vads-u-margin-top--0">
        Do you have a service-connected disability or a pre-discharge disability rating?
      </h5>

```

### After
```html
<h2 class="usa-alert-heading vads-u-font-size--h3">
  Save time—and save your work in progress—by signing in before starting your application
</h2>

<h2 class="vads-u-font-size--h3">
  Follow the steps below to apply for Planning and career guidance.
</h2>

  <h3 class="vads-u-font-size--h4">
    Prepare
  </h3>

    <h4 class="vads-u-font-size--h6">
      To fill out this application, you’ll need your:
    </h4>

  <h3>
    Apply
  </h3>

  <h3>
    VA Review
  </h3>

  <h3>
    Decision
  </h3>

<h2 class="vads-u-font-size--h3 vads-u-margin-top--0">
  Do you have a service-connected disability or a pre-discharge disability rating?
</h2>

```

## WCAG or Vendor Guidance (optional)

[axe-core 3.4 - Heading levels should only increase by one](https://dequeuniversity.com/rules/axe/3.4/heading-order)

## Screenshots or Trace Logs

The only visual change will be for the heading "Follow the steps below to apply for Planning and career guidance."

### Before
<img width="708" alt="Screen Shot 2020-11-30 at 5 13 51 PM" src="https://user-images.githubusercontent.com/14154792/100672086-6c657500-332f-11eb-899a-8299df66b8cc.png">
<img width="1717" alt="Screen Shot 2021-02-01 at 11 40 25 AM" src="https://user-images.githubusercontent.com/14154792/106489189-7e27dd80-6482-11eb-90ae-77f6c5db589f.png">

### After
<img width="700" alt="Screen Shot 2020-11-30 at 5 13 26 PM" src="https://user-images.githubusercontent.com/14154792/100672055-5f488600-332f-11eb-996d-b13c5bca3e9e.png">

**Comment by jason-gcio** [comment]
We will adjust this - moved to Ready column

**Comment by cohnjesse** [comment]
@joshkimux The H3 that is on this intro page is coming from the save-in-progress component that is built by the platform so it isn't something we can change. What if instead of doing H2 H2 H3 like your example we did H3 H3 H4?

**Comment by joshkimux** [comment]
@cohnjesse Thanks for the update! We could definitely give that a shot! Alternatively, we could also try to add a sr-only `h2` above the alert titled "alert," which would let us keep the h2 h2 h3 suggestion. It could also reduce the total rework in the future should the alert's `h3` be editable. What are your thoughts?

**Comment by cohnjesse** [comment]
@joshkimux I usually don't like adding screen reader only HTML, it seems like a misrepresentation of the page to me. Doing the alternative of H3 H3 H4 would be a lower impact on me but what do you think?

**Comment by joshkimux** [comment]
@cohnjesse I agree the alternative of H3 H3 H4 would be lower impact, but it won't be able to resolve the AXE issue (see below) that's thrown as no H2 will exist which causes the jump in order.

<img width="1731" alt="Screen Shot 2021-01-04 at 11 02 16 AM" src="https://user-images.githubusercontent.com/14154792/103554044-4fffaf80-4e7c-11eb-999f-8e182def8b38.png">

I also see your point on misrepresenting the page by adding a sr-only, and it's not ideal. The ideal solution would be to update the save in progress component's H3 to an H2 along with the following headings listed in the ticket OR to change the save in progress component's H3 to maybe a dfn or paragraph.

That being said, since our options are limited because of platform's save-in-progress component, maybe we could start by documenting this and tagging the platform team? I'm sorry about the roundabout approach this is creating 🙇

**Comment by cohnjesse** [comment]
@joshkimux My only concern with making this an H2 across the board at the platform level is that there are probably applications out there where having this be an H3 is actually fitting into a correct header progression. How about this - What if I make the adjustments to the platform component so that the developer can specify what he wants the header tag to be, be it an H2 or H3 or whatever, that way we won't break existing applications but rather just add the option for us and whoever else to specific if they want the header to be an H2?

**Comment by joshkimux** [comment]
@cohnjesse Absolutely agree with your point on the risk of making it an `h2` across the board. Your suggested solution sounds like a great way of approaching this! Before I sign off or commit on anything, I think @1Copenut would probably have thoughts / valuable historical knowledge on this as well.

**Comment by 1Copenut** [comment]
> What if I make the adjustments to the platform component so that the developer can specify what he wants the header tag to be

This to me is the right answer and supports long-term viability. We won't always know where in the hierarchy these information cards are being added, so it's beneficial to have a prop that can be set to override defaults. We're doing something very similar in the Alertbox component here: https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/4b63889c81be4f3e9036a4b93f1ad47424b45b8d/packages/formation-react/src/components/AlertBox/AlertBox.jsx#L74

**Comment by joshkimux** [comment]
Looks like we have our collective answer then!
Thank you @cohnjesse for the idea in the first place 🎉

**Comment by cohnjesse** [comment]
@joshkimux @1Copenut Ok, I will put together a separate ticket and do the work to add this functionality into the design system. Thanks!

**Comment by joshkimux** [comment]
<img width="1333" alt="Screen Shot 2021-02-01 at 11 07 30 AM" src="https://user-images.githubusercontent.com/14154792/106484697-df00e700-647d-11eb-9bd0-e396db9aadc3.png">

Reopening this ticket for the full audit as it has yet to be addressed in the save-in-progress component 🙇

**Comment by joshkimux** [comment]
<img width="1717" alt="Screen Shot 2021-02-01 at 11 40 25 AM" src="https://user-images.githubusercontent.com/14154792/106489225-88e27280-6482-11eb-86d2-453a4cddb6ba.png">

This also remains unfixed!

**Comment by jason-gcio** [comment]
Presented this to platform, was advised to leave open for now.
https://dsva.slack.com/archives/CBU0KDSB1/p1611789311332100

**Comment by jason-gcio** [comment]
These items are in a platform component and we cannot change them directly.

**Comment by micahchiang** [comment]
@jason-gcio - we can update the header of the blue rated disabilities section though. Perhaps a separate ticket should be made for that so it doesn't bet buried in here, which is what I think happened last time.

**Comment by jason-gcio** [comment]
@micahchiang - Ah ok, my misunderstanding.  I will write a ticket for that and link it here.

**Comment by joshkimux** [comment]
Thank you both!!

**Comment by micahchiang** [comment]
There hasn't been much movement on this ticket and it's been several months. We might want to consider just handling the update to the save in progress component ourselves. Doing so would also unblock #19755
